internal/test/e2e/run.go

```diff
@@ -204,19 +206,30 @@ func splitTests(testsList []string, conf ParallelRunConf) []instanceRunConf {
 
 	vsphereTestsRe := regexp.MustCompile(vsphereRegex)
 	privateNetworkTestsRe := regexp.MustCompile(`^.*(Proxy|RegistryMirror).*$`)
+	multiClusterTestsRe := regexp.MustCompile(`^.*Multicluster.*$`)
 
 	runConfs := make([]instanceRunConf, 0, conf.MaxInstances)
 	ipman := newE2EIPManager(os.Getenv(cidrVar), os.Getenv(privateNetworkCidrVar))
 
 	testsInCurrentInstance := make([]string, 0, testPerInstance)
 	for _, testName := range testsList {
 		testsInCurrentInstance = append(testsInCurrentInstance, testName)
-		ip := ""
+		multiClusterTest := multiClusterTestsRe.MatchString(testName)
+		var ips framework.E2EIPPool
 		if privateNetworkTestsRe.MatchString(testName) {
-			ip = ipman.getPrivateIP()
+			if multiClusterTest {
+				ips = ipman.reservePrivateIPPool(2)
```

Comment:
We don't have any multi cluster private network tests at the moment, but I added just in case it's something we require in the future.

Also, as I commented previously I don't love just hard coding pool count to 1 or 2, but for now I think it will do. As I mention in previous comment I'm open to solutions on how we can account for the count on a test by test basis.